### PR TITLE
Docs/update button component props

### PR DESCRIPTION
**Pull request by mateuswetah**
## Description
Added missing props to the Button component props list, such as `className`, `target`, `icon`, `showTooltip` and others related. Also, shows the props list more alike how it is done in other components: a list instead of a table.

## Types of changes
Documentation improvements

### CODE REVIEW

File: packages/components/src/button/README.md [hunk 1]
```diff
@@ -15,19 +16,19 @@ Buttons let users take actions and make choices with a single click or tap.
 
 Buttons tell users what actions they can take and give them a way to interact with the interface. You’ll find them throughout a UI, particularly in places like:
 
-- Modals
-- Forms
-- Toolbars
+-   Modals
+-   Forms
+-   Toolbars
```
Comment (youknowriad) [inline packages/components/src/button/README.md:21]: Why all lists are not formatted differently?

**Review by youknowriad — COMMENTED** [review]
Seems like a nice addition, I don't understand why the formatting change is needed though.

File: packages/components/src/button/README.md (hunk 1 above)
Comment (mateuswetah) [reply packages/components/src/button/README.md:21]: Hi @youknowriad. I am not sure why, but after installing the prettier extension as in [here](https://developer.wordpress.org/block-editor/contributors/document/#editor-config), every time I saved my file the one space that I inserted was converted to two spaces ¬¬'. Is there anything that I should change in my pretter settings?

Comment (youknowriad) [reply packages/components/src/button/README.md:21]: I think it's probably fine because we don't have strong linting checks for markdown. 👍

**Review by youknowriad — APPROVED** [review]

**Pull request merged**